tests/unit/copying_manager_tests/copying_manager_test.py

```diff
@@ -48,898 +43,174 @@
-            self.config = config
-            self.log_path = config["path"]
+        if root_dir:
+            return manager
 
-    class MonitorObject(object):
-        def __init__(self, config):
-            self.module_name = config["module"]
-            self.config = config
-            self.log_config = {"path": self.module_name.split(".")[-1] + ".log"}
+        self._log_file1 = files[0]
+        self.__test_log_file = self._log_file1.str_path
 
-    def __create_test_configuration_instance(self):
+        self._config = self._env_builder.config
 
-        default_paths = DefaultPaths(
-            "/var/log/scalyr-agent-2",
-            "/etc/scalyr-agent-2/agent.json",
-            "/var/lib/scalyr-agent-2",
-        )
-        return TestingConfiguration(self.__config_file, default_paths, None)
-
-
-class CopyingParamsTest(ScalyrTestCase):
-    def setUp(self):
-        super(CopyingParamsTest, self).setUp()
-        self.__config_dir = tempfile.mkdtemp()
-        self.__config_file = os.path.join(self.__config_dir, "agentConfig.json")
-        self.__config_fragments_dir = os.path.join(self.__config_dir, "configs.d")
-        os.makedirs(self.__config_fragments_dir)
-
-        fp = open(self.__config_file, "w")
-        fp.write('{api_key: "fake"}')
-        fp.close()
-
-        config = self.__create_test_configuration_instance()
-        config.parse()
-        self.test_params = CopyingParameters(config)
-
-    def test_initial_settings(self):
-        self.assertEquals(self.test_params.current_bytes_allowed_to_send, ALMOST_SIX_MB)
-        self.assertEquals(self.test_params.current_sleep_interval, 5.0)
-
-    def test_no_events_being_sent(self):
-        for i in range(0, 5):
-            self.test_params.update_params("success", 0)
-            self.assertEquals(
-                self.test_params.current_bytes_allowed_to_send, ALMOST_SIX_MB
-            )
-            self.assertEquals(self.test_params.current_sleep_interval, 5.0)
-
-    def test_small_events_being_sent(self):
-        self.test_params.current_sleep_interval = 0.1
-        self._run(
-            "success",
-            10 * 1024,
-            [0.15, ALMOST_SIX_MB],
-            [0.225, ALMOST_SIX_MB],
-            [0.3375, ALMOST_SIX_MB],
-            [0.50625, ALMOST_SIX_MB],
-        )
+        return manager
 
-    def test_too_many_events_being_sent(self):
-        self.test_params.current_sleep_interval = 1
+    def __append_log_lines(self, *lines):
+        self._log_file1.append_lines(*lines)
 
-        self._run(
-            "success",
-            200 * 1024,
-            [0.6, ALMOST_SIX_MB],
-            [0.36, ALMOST_SIX_MB],
-            [0.216, ALMOST_SIX_MB],
-            [0.1296, ALMOST_SIX_MB],
-        )
+    def __extract_lines(self, request):
+        return self._extract_lines(request)
 
-    def test_request_too_big(self):
-        self.test_params.current_sleep_interval = 1
+    def assertEquals(self, expected, actual):
```

Comment:
This was made to be able to run those tests on `pytest`  but without spending time on rewriting the tests cases code.
The reason for that is that the `pytest` is able to run `unittest` test cases but it can not parameterize them.
I've also left the TODO to rewrite it and to remove those methods.
https://github.com/scalyr/scalyr-agent-2/blob/91aa95fb47e1120aef1fb4f4dac487599285aff9/tests/unit/copying_manager_tests/copying_manager_test.py#L104-L114